vra/resource_block_device.go

```diff
@@ -249,7 +251,32 @@ func resourceBlockDeviceRead(d *schema.ResourceData, m interface{}) error {
 
 func resourceBlockDeviceUpdate(d *schema.ResourceData, m interface{}) error {
 
-	return fmt.Errorf("Updating a block device resource is not allowed")
+	log.Printf("Starting to update the vra_block_device resource with name %s", d.Get("name"))
+	apiClient := m.(*Client).apiClient
+
+	id := d.Id()
+	resp, err := apiClient.Disk.GetBlockDevice(disk.NewGetBlockDeviceParams().WithID(id))
+	if err != nil {
+		return err
+	}
+	capacityInGB := int32(d.Get("capacity_in_gb").(int))
+	inputMap := make(map[string]interface{})
+	inputMap["capacityGB"] = capacityInGB
+
+	deploymentID := resp.GetPayload().DeploymentID
```

Comment:
Since this is pulling a value out of the GetBlockDevice() response, move it above the capacityInGB line and closer to the GetBlockDevice() call.

Reply:
Done!